server/http4s-server/src/main/scala/sttp/tapir/server/http4s/TapirHttp4sServer.scala

```diff
@@ -12,6 +13,30 @@ import sttp.tapir.typelevel.ReplaceFirstInTuple
 import scala.reflect.ClassTag
 
 trait TapirHttp4sServer {
+  implicit class RichHttp4sHttpEndpoint0[I, E, O, F[_], G[_]](e: Endpoint[I, E, O, EntityBody[F]]) {
```

Comment:
Since `G` is free here (not restricted by `e`), it should be a parameter in `toHttp` and `toHttpRecoverErrors`, both of which we could then move to `RichHttp4sHttpEndpoint`.